src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -104,9 +104,10 @@ internal static partial class ExpressionSyntaxExtensions
             CancellationToken cancellationToken)
         {
             if (targetType.ContainsAnonymousType())
-            {
                 return expression;
-            }
+
+            if (targetType.IsSystemVoid())
+                return expression;
```

Comment:
this prevent other parts of the system from adding `(void)` casts which were very much throwing off other components.